finagle-http/src/main/scala/com/twitter/finagle/http/filter/ComplianceFilter.scala

```diff
@@ -0,0 +1,62 @@
+package com.twitter.finagle.http.filter
+
+import com.twitter.finagle.http.{Fields, Request, Response, Status}
+import com.twitter.finagle._
+import com.twitter.logging.Logger
+import com.twitter.util.Future
+import org.jboss.netty.buffer.ChannelBuffers
+
+/**
+ * Modify a content and header field filter.
+ *
+ * @see [[https://tools.ietf.org/html/rfc2616 "Hypertext Transfer Protocol -- HTTP/1.1"]]
+ * @see [[https://tools.ietf.org/html/rfc7230 "Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing"]]
+ * section-3.3.2 on `Content-Length` for more information.
+ */
+private[finagle] class ComplianceFilter[Req <: Request] extends SimpleFilter[Req, Response] {
+  import ComplianceFilter._
+  val logger = Logger(getClass)
+
+  def apply(request: Req, service: Service[Req, Response]): Future[Response] = {
+    service(request).map { rep =>
+      if (mustNotIncludeContent(rep.status)) {
+        if (rep.getContent() != ChannelBuffers.EMPTY_BUFFER) {
+          rep.clearContent()
+          logger.warning(s"Response with a status code of ${rep.statusCode} must not have a body-message thus the content has been removed.")
+        }
+
+        if (rep.status == Status.NoContent && rep.contentLength.isDefined) {
+          rep.headerMap.remove(Fields.ContentLength)
+          logger.warning(s"Response with a status code of ${rep.statusCode} must not have a Content-Length header field thus the field has been removed.")
+        }
+
+        //Make sure a Content-Length header field is set with a value of 0 if it is explicitly sent for a 304 response
```

Comment:
Oof, this doesn't feel like it's easy to know.  I guess this is something that users must set?  I guess it's better to just not modify the content length in this case.